tm2/pkg/amino/pkg/pkg.go

```diff
@@ -196,6 +201,59 @@ func (pkg *Package) WithP3SchemaFile(file string) *Package {
 	return pkg
 }
 
+// Parse the Go code in filename and scan the AST looking for struct doc comments.
+// Find the Type in pkg.Types and set its Comment and FieldComments, which are
+// used by genproto.GenerateProto3MessagePartial to set the Comment in the P3Doc
+// and related P3Field objects.
+func (pkg *Package) WithComments(filename string) *Package {
+	fset := token.NewFileSet()
+	f, err := parser.ParseFile(fset, filename, nil, parser.ParseComments)
+	if err != nil {
+		panic(err)
+	}
+
+	ast.Inspect(f, func(node ast.Node) bool {
+		if genDecl, ok := node.(*ast.GenDecl); ok {
+			for _, spec := range genDecl.Specs {
+				if typeSpec, ok := spec.(*ast.TypeSpec); ok {
+					if pkgType := pkg.getTypeByName(typeSpec.Name.Name); pkgType != nil {
+						if genDecl.Doc != nil {
+							// Set the type comment.
+							pkgType.Comment = strings.TrimSpace(genDecl.Doc.Text())
+						}
+						if structType, ok := typeSpec.Type.(*ast.StructType); ok {
+							for _, field := range structType.Fields.List {
+								if field.Names != nil && len(field.Names) == 1 && field.Doc != nil {
+									// Set the field comment.
+									if pkgType.FieldComments == nil {
```

Comment:
Actually, let's flatten this mess instead of wrapping it with 8 levels of indentation.

Ie. `if genDecl, ok := node.(*ast.GenDecl); ok {` becomes:

```go
genDecl, ok := node.(*ast.GenDecl)
if !ok {
    return true
}
```

and so on for the other assertions, including the for loop where we use a `continue` instead.